modules/storage-page-memory/src/test/java/org/apache/ignite/internal/storage/pagememory/index/AbstractPageMemoryHashIndexStorageTest.java

```diff
@@ -0,0 +1,132 @@
+ */
+abstract class AbstractPageMemoryHashIndexStorageTest extends AbstractHashIndexStorageTest {
+    protected BasePageMemoryStorageEngineConfiguration<?, ?> baseEngineConfig;
+
+    /**
+     * Initializes the internal structures needed for tests.
+     *
+     * <p>This method *MUST* always be called in either subclass' constructor or setUp method.
+     */
+    protected final void initialize(
+            MvTableStorage tableStorage,
+            TablesConfiguration tablesCfg,
+            BasePageMemoryStorageEngineConfiguration<?, ?> baseEngineConfig
+    ) {
+        this.baseEngineConfig = baseEngineConfig;
+
+        initialize(tableStorage, tablesCfg);
+    }
+
+    //TODO IGNITE-17626 Enable the test.
+    @Disabled
+    @Override
+    public void testDestroy() {
+    }
+
+    @Test
+    void testWithHashCollisionStrings() {
+        String[] hashCollisionPairStrings = createHashCollisionPairStrings("foo");
+
+        IndexRow indexRow0 = createIndexRow(1, hashCollisionPairStrings[0], new RowId(TEST_PARTITION));
+        IndexRow indexRow1 = createIndexRow(1, hashCollisionPairStrings[1], new RowId(TEST_PARTITION));
+
+        put(indexRow0);
+        put(indexRow1);
+
+        assertThat(getAll(indexRow0), contains(indexRow0.rowId()));
+        assertThat(getAll(indexRow1), contains(indexRow1.rowId()));
+    }
+
+    @Test
+    void testWithStringsLargerThanMaximumInlineSize() {
+        String[] hashCollisionPairStrings = createHashCollisionPairStrings(randomString(MAX_BINARY_TUPLE_INLINE_SIZE));
+
+        IndexRow indexRow0 = createIndexRow(1, hashCollisionPairStrings[0], new RowId(TEST_PARTITION));
+        IndexRow indexRow1 = createIndexRow(1, hashCollisionPairStrings[1], new RowId(TEST_PARTITION));
+
+        put(indexRow0);
+        put(indexRow1);
+
+        assertThat(getAll(indexRow0), contains(indexRow0.rowId()));
+        assertThat(getAll(indexRow1), contains(indexRow1.rowId()));
+
+        assertThat(getAll(createIndexRow(1, "foo", new RowId(TEST_PARTITION))), empty());
+    }
+
+    @Test
+    void testFragmentedIndexColumns() {
+        String[] hashCollisionPairStrings = createHashCollisionPairStrings(randomString(baseEngineConfig.pageSize().value() * 2));
+
+        IndexRow indexRow0 = createIndexRow(1, hashCollisionPairStrings[0], new RowId(TEST_PARTITION));
+        IndexRow indexRow1 = createIndexRow(1, hashCollisionPairStrings[1], new RowId(TEST_PARTITION));
+
+        put(indexRow0);
+        put(indexRow1);
+
+        assertThat(getAll(indexRow0), contains(indexRow0.rowId()));
+        assertThat(getAll(indexRow1), contains(indexRow1.rowId()));
+
+        assertThat(getAll(createIndexRow(1, "foo", new RowId(TEST_PARTITION))), empty());
+    }
+
+    private static String[] createHashCollisionPairStrings(String prefix) {
+        String s0 = prefix + "Aa";
+        String s1 = prefix + "BB";
+
+        assertEquals(s0.hashCode(), s1.hashCode(), "s0=" + s0 + ", s1=" + s1);
```

Comment:
Very nice!

Reply:
I removed it because the hash code for the tuple is calculated differently and this does not suit us.